charts/pact-broker/templates/_helpers.tpl

```diff
@@ -158,7 +158,16 @@ Database ENV Vars
 - name: PACT_BROKER_DATABASE_NAME
   value: {{ include "broker.databaseName" . }}
 - name: PACT_BROKER_DATABASE_USERNAME
-  value: {{ include "broker.databaseUser" . }}
+  {{- if .Values.postgresql.enabled  }}
+  value: {{ .Values.postgresql.auth.password | quote }}
+  {{- else if and .Values.externalDatabase.enabled .Values.externalDatabase.config.auth.username }}
+  value: {{ .Values.externalDatabase.config.auth.username | quote }}
+  {{- else }}
+  valueFrom:
+    secretKeyRef:
+      name: {{ include "broker.databaseUserSecretName" . }}
```

Comment:
`databaseUserSecretName` or `databaseUserSecretKey` isn't defined anywhere in the tmpl file, you'll have to add it if you intend on using it